networkit/cpp/planarity/LeftRightPlanarityCheck.cpp

```diff
@@ -0,0 +1,262 @@
+        auto highestReturnEdgeLeft = stack.top().left.high;
+        auto highestReturnEdgeRight = stack.top().right.high;
+        if (highestReturnEdgeLeft != noneEdge
+            && (highestReturnEdgeRight != noneEdge
+                || lowestPoint[highestReturnEdgeLeft] > lowestPoint[highestReturnEdgeRight])) {
+            ref[edge] = highestReturnEdgeLeft;
+        } else {
+            ref[edge] = highestReturnEdgeRight;
+        }
+    }
+}
+
+bool LeftRightPlanarityCheck::dfsTesting(node startNode) {
+    std::stack<node> dfsStack;
+    dfsStack.emplace(startNode);
+    auto neighborIterators =
+        std::unordered_map<node, decltype(dfsGraph.neighborRange(startNode).begin())>{};
+
+    auto preprocessedEdges = std::unordered_set<Edge>{};
+    while (!dfsStack.empty()) {
+        const auto currentNode = dfsStack.top();
+        dfsStack.pop();
+        const auto parentEdge = parentEdges[currentNode];
+        bool callRemoveBackEdges{true};
+        if (auto it = neighborIterators.find(currentNode); it == neighborIterators.end()) {
+            neighborIterators[currentNode] = dfsGraph.neighborRange(currentNode).begin();
+        }
+        auto &neighborIterator = neighborIterators[currentNode];
+        while (neighborIterator != dfsGraph.neighborRange(currentNode).end()) {
+            const auto neighbor = *neighborIterator;
+            auto currentEdge = Edge(currentNode, neighbor);
+            if (!preprocessedEdges.contains(currentEdge)) {
+                stackBottom[currentEdge] = stack.empty() ? NoneConflictPair : stack.top();
+                if (currentEdge == parentEdges[neighbor]) {
+                    dfsStack.emplace(currentNode);
+                    dfsStack.emplace(neighbor);
+                    preprocessedEdges.insert(currentEdge);
+                    callRemoveBackEdges = false;
+                    break;
+                }
+                lowestPointEdge[currentEdge] = currentEdge;
+                stack.emplace(Interval{}, Interval(currentEdge, currentEdge));
+            }
+
+            if (lowestPoint.contains(currentEdge)
+                && lowestPoint[currentEdge] < heights[currentNode]) {
+                if (neighbor == *dfsGraph.neighborRange(currentNode).begin()) {
+                    lowestPointEdge[parentEdge] = lowestPointEdge[currentEdge];
+                } else {
+                    if (!applyConstraints(currentEdge, parentEdge))
+                        return false;
+                }
+            }
+            ++neighborIterator;
+        }
+
+        if (callRemoveBackEdges) {
+            if (parentEdge != noneEdge)
+                removeBackEdges(parentEdge);
+        }
+    }
+    return true;
+}
+
+void LeftRightPlanarityCheck::dfsOrientation(const node startNode) {
+
+    std::stack<node> dfsStack;
+    dfsStack.emplace(startNode);
+    auto preprocessedEdges = std::unordered_set<Edge>{};
+    while (!dfsStack.empty()) {
```

Comment:
As above, replace with `do {} while` and avoid using `auto` for non-iterator types.